File: client/src/app/modules/quotation/services/provision.screen.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { ProvisionScreenType } from '../../quotation/model/ProvisionScreenType';

@Injectable({
  providedIn: 'root'
})
export class ProvisionScreenTypeService extends AppRestService<ProvisionScreenType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getProvisionScreenTypes() {
    return this.getListCached(new HttpParams(), "provision-screen-types");
  }

  addOrUpdateProvisionScreenType(provisionScreenType: ProvisionScreenType) {
    this.clearListCache(new HttpParams(), "provision-screen-types");
    return this.addOrUpdate(new HttpParams(), "provision-screen-type", provisionScreenType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/ProvisionScreenTypeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.ProvisionScreenType;

public interface ProvisionScreenTypeService {
    public List<ProvisionScreenType> getProvisionScreenTypes();

    public ProvisionScreenType getProvisionScreenType(Integer id);

    public void updateScreenTypes();

}
